EngineLayer/Psm.cs

```diff
@@ -267,15 +267,25 @@ public void AddOrReplace(CompactPeptideBase compactPeptide, double score, int no
 
         public void MatchToProteinLinkedPeptides(Dictionary<CompactPeptideBase, HashSet<PeptideWithSetModifications>> matching)
         {
-            foreach (var ok in compactPeptides.Keys.ToList())
+            foreach (var cpKey in compactPeptides.Keys.ToList())
             {
-                compactPeptides[ok] = new Tuple<int, HashSet<PeptideWithSetModifications>>(compactPeptides[ok].Item1, matching[ok]);
-
-                var candidatePli = new ProteinLinkedInfo(matching[ok]);
+                compactPeptides[cpKey] = new Tuple<int, HashSet<PeptideWithSetModifications>>(compactPeptides[cpKey].Item1, matching[cpKey]);
+                var candidatePli = new ProteinLinkedInfo(matching[cpKey]);
+                if (MostProbableProteinInfo == null || FirstIsPreferable(candidatePli, MostProbableProteinInfo))
+                    MostProbableProteinInfo = candidatePli;
+            }
+            List<CompactPeptideBase> duplicateKeysToRemove = new List<CompactPeptideBase>();
+            foreach (KeyValuePair<CompactPeptideBase, Tuple<int, HashSet<PeptideWithSetModifications>>> kvp in compactPeptides)
+                if (kvp.Value.Item2 == null)
```

Comment:
This should have never been null in the first place. If this condition is true it means there is a bug somewhere before this point.